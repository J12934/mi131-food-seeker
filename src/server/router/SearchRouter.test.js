const SearchRouter = require('./SearchRouter');

jest.mock('../services/YelpAPI');

describe('SearchRouter', () => {
    it('should emit a new event when the search succeds', async () => {
        const wsMock = new class {
            emit = jest.fn();
        }();

        const searchRouter = new SearchRouter({ client: wsMock });

        const response = await searchRouter.query({
            term: 'burger',
            location: 'Kiel',
        });

        expect(wsMock.emit).toBeCalled();
        expect(wsMock.emit.mock.calls).toMatchSnapshot();
    });
});
