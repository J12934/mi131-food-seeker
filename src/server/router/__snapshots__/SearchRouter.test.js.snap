// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchRouter should emit a new event when the search succeds 1`] = `
Array [
  Array [
    "results",
    Object {
      "search": Object {
        "business": Array [
          Object {
            "name": "resturant 1",
          },
        ],
      },
    },
  ],
]
`;
